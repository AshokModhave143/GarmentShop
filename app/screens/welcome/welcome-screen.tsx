import React from 'react'
import { View, ViewStyle, TextStyle, SafeAreaView } from 'react-native'
import { useNavigation } from '@react-navigation/native'
import { observer } from 'mobx-react-lite'
import { Button, Screen, Text } from '../../components'
import { color, spacing, typography } from '../../theme'
import { useTranslation } from 'react-i18next'

const FULL: ViewStyle = { flex: 1 }
const CONTAINER: ViewStyle = {
  backgroundColor: color.transparent,
  paddingHorizontal: spacing[4],
}
const TEXT: TextStyle = {
  color: color.palette.white,
  fontFamily: typography.primary,
}
const BOLD: TextStyle = { fontWeight: 'bold' }
const TITLE_WRAPPER: TextStyle = {
  ...TEXT,
  textAlign: 'center',
}
const TITLE: TextStyle = {
  ...TEXT,
  ...BOLD,
  fontSize: 28,
  lineHeight: 38,
  textAlign: 'center',
}
const ALMOST: TextStyle = {
  ...TEXT,
  ...BOLD,
  fontSize: 26,
  fontStyle: 'italic',
}
const CONTENT: TextStyle = {
  ...TEXT,
  color: '#BAB6C8',
  fontSize: 15,
  lineHeight: 22,
  marginBottom: spacing[5],
}
const CONTINUE: ViewStyle = {
  paddingVertical: spacing[4],
  paddingHorizontal: spacing[4],
  backgroundColor: color.palette.deepPurple,
}
const CONTINUE_TEXT: TextStyle = {
  ...TEXT,
  ...BOLD,
  fontSize: 13,
  letterSpacing: 2,
}
const FOOTER: ViewStyle = { backgroundColor: '#20162D' }
const FOOTER_CONTENT: ViewStyle = {
  paddingVertical: spacing[4],
  paddingHorizontal: spacing[4],
}

export const WelcomeScreen = observer(function WelcomeScreen() {
  const navigation = useNavigation()
  const nextScreen = () => navigation.navigate('dashboard')
  const { t } = useTranslation()

  return (
    <View testID="WelcomeScreen" style={FULL}>
      <Screen style={CONTAINER} preset="scroll" backgroundColor={color.transparent}>
        <Text style={TITLE_WRAPPER}>
          <Text style={TITLE} text="Your new app, " />
          <Text style={ALMOST} text="almost" />
          <Text style={TITLE} text="!" />
        </Text>

        <Text style={CONTENT}>
          This probably isn't what your app is going to look like. Unless your designer handed you
          this screen and, in that case, congrats! You're ready to ship.
        </Text>
      </Screen>
      <SafeAreaView style={FOOTER}>
        <View style={FOOTER_CONTENT}>
          <Button
            testID="next-screen-button"
            style={CONTINUE}
            textStyle={CONTINUE_TEXT}
            tx="welcomeScreen.continue"
            onPress={nextScreen}
          />
        </View>
      </SafeAreaView>
    </View>
  )
})
