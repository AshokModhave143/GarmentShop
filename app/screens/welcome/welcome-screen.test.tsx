import React from 'react'
import { render, cleanup } from '../../../test/test-util'
import { WelcomeScreen } from './welcome-screen'

afterEach(cleanup)

describe('Welcome screen', () => {
  it('should render the component without crashing', () => {
    const { toJSON } = render(<WelcomeScreen />, {})
    expect(toJSON()).toMatchSnapshot()
  })
})
