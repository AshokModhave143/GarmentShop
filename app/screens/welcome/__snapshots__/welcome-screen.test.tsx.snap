// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome screen should render the component without crashing 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
    testID="WelcomeScreen"
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "flex": 1,
              "height": "100%",
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
            },
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "flex": 1,
              "height": "100%",
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
            },
            Object {
              "paddingTop": 0,
            },
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {
                "alignItems": "stretch",
                "justifyContent": "flex-start",
              },
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0)",
                "paddingHorizontal": 16,
              },
            ]
          }
          keyboardShouldPersistTaps="handled"
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "flex": 1,
                "height": "100%",
              },
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0)",
              },
            ]
          }
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Helvetica",
                    "fontSize": 15,
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Helvetica",
                    "textAlign": "center",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Helvetica",
                      "fontSize": 15,
                    },
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Helvetica",
                      "fontSize": 28,
                      "fontWeight": "bold",
                      "lineHeight": 38,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Your new app, 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Helvetica",
                      "fontSize": 15,
                    },
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Helvetica",
                      "fontSize": 26,
                      "fontStyle": "italic",
                      "fontWeight": "bold",
                    },
                  ]
                }
              >
                almost
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Helvetica",
                      "fontSize": 15,
                    },
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Helvetica",
                      "fontSize": 28,
                      "fontWeight": "bold",
                      "lineHeight": 38,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                !
              </Text>
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Helvetica",
                    "fontSize": 15,
                  },
                  Object {
                    "color": "#BAB6C8",
                    "fontFamily": "Helvetica",
                    "fontSize": 15,
                    "lineHeight": 22,
                    "marginBottom": 24,
                  },
                ]
              }
            >
              This probably isn't what your app is going to look like. Unless your designer handed you this screen and, in that case, congrats! You're ready to ship.
            </Text>
          </View>
        </RCTScrollView>
      </View>
    </View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "#20162D",
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#5D2555",
              "borderRadius": 4,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
          testID="next-screen-button"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Helvetica",
                  "fontSize": 15,
                },
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 9,
                    "paddingHorizontal": 12,
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Helvetica",
                    "fontSize": 13,
                    "fontWeight": "bold",
                    "letterSpacing": 2,
                  },
                ],
              ]
            }
          >
            welcomeScreen.continue.test
          </Text>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</RNCSafeAreaProvider>
`;
